### ISSUE TITLE
drush permission denied

### ISSUE THREAD

**Issue by rroblik** [wodby/docker4drupal#261]
### Codebase
Built-in vanilla Drupal
  
### Host OS 
Debian 8

### Logs output
```
...
 [warning] file_put_contents(/home/wodby/.drush/cache/factory/Drush.Drupal.Commands.sql.SanitizeUserTableCommands.cache): failed to open stream: Permission denied JSONCache.php:30
...
```

Hi,

When running (any) drush command in php container (`docker exec -ti myproject_php bash
`), I have a lot of warning messages, related to permission error for `drush`.

Can you please check these?

Regards

**Comment by csandanov** [comment]
Can you show the output of:

```shell
# enter php container
$ make shell
# from php container:
$ id
$ ls -la /home/wodby/.drush
```

**Comment by rroblik** [comment]
Here is the output

```bash
deuser@vm-localdomain ~ $ docker exec -ti 3d_php bash
/var/www/html$ make shell
make: *** No rule to make target 'shell'.  Stop.
/var/www/html$ id
uid=1000(wodby) gid=1000(wodby) groups=82(www-data),1000(wodby)
/var/www/html$ ls -la /home/wodby/.drush
total 20
drwxrwxr-x    8 wodby    wodby         4096 Mar 25 19:13 .
drwxrwxr-x   11 wodby    wodby         4096 Mar 25 20:10 ..
drwxrwxr-x    7 wodby    wodby         4096 Mar 25 19:13 cache
drwxrwxr-x    6 wodby    wodby         4096 Mar 25 19:13 drush-patchfile
drwxrwxr-x    2 wodby    wodby         4096 Mar 25 19:13 registry_rebuild
```

**Comment by csandanov** [comment]
Well, permissions seem to be correct. How do you run your drush command? Check the contents of `/home/wodby/.drush/cache/factory/` and make sure everything owned by `wodby:wodby`.

**Comment by rroblik** [comment]
Hi,
Drush command is 
(host) : `docker exec -ti 3d_php bash`
(in container)  : `drush st`

**Comment by csandanov** [comment]
I can't find any explanations why these files owned by root other than you run drush commands from root user. 

```shell
$ docker run --rm -ti wodby/drupal:8-7.1-4.2.2 sh                                                                                          [21:06:22]
Drupal 8 not found in /var/www/html - copying now...
Complete! Drupal 8 has been successfully copied to /var/www/html
/var/www/html $ cd web/
/var/www/html/web $ drush st
 Drupal version : 8.5.0
 Site URI       : default
 DB driver      : mysql
 DB hostname    : mariadb
 DB port        :
 DB username    : drupal
 DB name        : drupal
 PHP binary     : /usr/local/bin/php
 PHP config     :
 PHP OS         : Linux
 Drush script   : /var/www/html/vendor/drush/drush/drush
 Drush version  : 9.2.1
 Drush temp     : /tmp
 Drush configs  : /var/www/html/vendor/drush/drush/drush.yml
                  /var/www/html/drush/drush.yml
 Drupal root    : /var/www/html/web
 Site path      : sites/default
/var/www/html/web $ ls -la ~/.drush/cache/factory/
total 216
drwxr-sr-x    2 wodby    wodby         4096 Mar 28 14:06 .
drwxr-sr-x    5 wodby    wodby         4096 Mar 28 14:06 ..
-rw-r--r--    1 wodby    wodby         3302 Mar 28 14:06 Drush.Commands.ExampleCommands.cache
-rw-r--r--    1 wodby    wodby         8924 Mar 28 14:06 Drush.Commands.LegacyCommands.cache
-rw-r--r--    1 wodby    wodby         5543 Mar 28 14:06 Drush.Commands.OptionsCommands.cache
-rw-r--r--    1 wodby    wodby         4443 Mar 28 14:06 Drush.Commands.ValidatorsCommands.cache
-rw-r--r--    1 wodby    wodby         4417 Mar 28 14:06 Drush.Commands.config.ConfigPullCommands.cache
-rw-r--r--    1 wodby    wodby         3765 Mar 28 14:06 Drush.Commands.core.BrowseCommands.cache
-rw-r--r--    1 wodby    wodby         9602 Mar 28 14:06 Drush.Commands.core.CacheCommands.cache
-rw-r--r--    1 wodby    wodby         5146 Mar 28 14:06 Drush.Commands.core.CoreCommands.cache
-rw-r--r--    1 wodby    wodby        10046 Mar 28 14:06 Drush.Commands.core.DocsCommands.cache
-rw-r--r--    1 wodby    wodby         3744 Mar 28 14:06 Drush.Commands.core.DrupalDirectoryCommands.cache
-rw-r--r--    1 wodby    wodby         2605 Mar 28 14:06 Drush.Commands.core.DrupliconCommands.cache
-rw-r--r--    1 wodby    wodby         3780 Mar 28 14:06 Drush.Commands.core.EditCommands.cache
-rw-r--r--    1 wodby    wodby         3268 Mar 28 14:06 Drush.Commands.core.InitCommands.cache
-rw-r--r--    1 wodby    wodby         4119 Mar 28 14:06 Drush.Commands.core.LoginCommands.cache
-rw-r--r--    1 wodby    wodby         2711 Mar 28 14:06 Drush.Commands.core.NotifyCommands.cache
-rw-r--r--    1 wodby    wodby         4711 Mar 28 14:06 Drush.Commands.core.PhpCommands.cache
-rw-r--r--    1 wodby    wodby         6368 Mar 28 14:06 Drush.Commands.core.RsyncCommands.cache
-rw-r--r--    1 wodby    wodby         4743 Mar 28 14:06 Drush.Commands.core.RunserverCommands.cache
-rw-r--r--    1 wodby    wodby         6961 Mar 28 14:06 Drush.Commands.core.SiteCommands.cache
-rw-r--r--    1 wodby    wodby         8157 Mar 28 14:06 Drush.Commands.core.SiteInstallCommands.cache
-rw-r--r--    1 wodby    wodby         3290 Mar 28 14:06 Drush.Commands.core.SshCommands.cache
-rw-r--r--    1 wodby    wodby         5826 Mar 28 14:06 Drush.Commands.core.StatusCommands.cache
-rw-r--r--    1 wodby    wodby         3471 Mar 28 14:06 Drush.Commands.core.TopicCommands.cache
-rw-r--r--    1 wodby    wodby         7758 Mar 28 14:06 Drush.Commands.core.UpdateDBCommands.cache
-rw-r--r--    1 wodby    wodby         3148 Mar 28 14:06 Drush.Commands.core.XhprofCommands.cache
-rw-r--r--    1 wodby    wodby         2932 Mar 28 14:06 Drush.Commands.generate.GenerateCommands.cache
-rw-r--r--    1 wodby    wodby         3314 Mar 28 14:06 Drush.Commands.help.HelpCommands.cache
-rw-r--r--    1 wodby    wodby         3236 Mar 28 14:06 Drush.Commands.help.ListCommands.cache
-rw-r--r--    1 wodby    wodby         3171 Mar 28 14:06 Drush.Commands.pm.SecurityUpdateCommands.cache
-rw-r--r--    1 wodby    wodby        11746 Mar 28 14:06 Drush.Commands.sql.SqlCommands.cache
-rw-r--r--    1 wodby    wodby         6275 Mar 28 14:06 Drush.Commands.sql.SqlSyncCommands.cache
```

**Comment by rroblik** [comment]
But I can't give chown "wodby" user because I've no root password ...

**Comment by csandanov** [comment]
Something isn't healthy with your container, cache files shouldn't be owned by root. Just recreate the php container.

**Comment by csandanov** [comment]
After all, there was a bug, we executed drush from root during the initial rsync for vanilla Drupal. Fixed now, available in non-tagged vanilla images for now. Thank you for reporting this issue!

**Comment by Medi4tor** [comment]
Hi @csandanov , wel done.
What can we do to not have this error anymore?
K.R.

**Comment by dshields4** [comment]
I'm also seeing this.
Unable to drush cc drush - any advice on how to resolve this?

**Comment by dshields4** [comment]
I was able to clear the drush caches after rebuilding my container after making the following change to docker-compose.override.yml:
-    image: wodby/drupal:$DRUPAL_TAG
+    image: wodby/drupal

**Comment by csandanov** [comment]
Fixed with [5.0.5](https://github.com/wodby/docker4drupal/releases/tag/5.0.5) update.
